src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextHost.WorkspaceProjectContextHostInstance.cs

```diff
@@ -103,7 +106,18 @@ protected override async Task InitializeCoreAsync(CancellationToken cancellation
                         _projectSubscriptionService.ProjectBuildRuleSource.SourceBlock.SyncLinkOptions(GetProjectBuildOptions()),
                         _projectBuildSnapshotService.SourceBlock.SyncLinkOptions(),
                             target: DataflowBlockFactory.CreateActionBlock<IProjectVersionedValue<ValueTuple<ConfiguredProject, IProjectSubscriptionUpdate, IProjectBuildSnapshot>>>(e =>
-                                OnProjectChangedAsync(e, evaluation: false),
+                                OnProjectChangedAsync(e, WorkspaceContextHandlerType.ProjectBuild),
+                                _project.UnconfiguredProject,
+                                ProjectFaultSeverity.LimitedFunctionality),
+                            linkOptions: DataflowOption.PropagateCompletion,
+                            cancellationToken: cancellationToken),
+
+                    ProjectDataSources.SyncLinkTo(
+                        _activeConfiguredProjectProvider.ActiveConfiguredProjectBlock.SyncLinkOptions(),
+                        _projectSubscriptionService.SourceItemsRuleSource.SourceBlock.SyncLinkOptions(),
+                        _projectBuildSnapshotService.SourceBlock.SyncLinkOptions(),
```

Comment:
This is wrong it should be not sync'd with the build snapshot, otherwise, the results wait until the design time build has finished. I know that evaluation does this but this is a regression that needs to be fixed, and might explain why things are ending up in misc on first open.

Reply:
I've opened https://github.com/dotnet/project-system/issues/7423 for evaluation.

Reply:
Thanks. This is so annoying to change 😛

We need generic generics so I can define a method that takes a tuple of any size as long as the first item is of a specific type :)